packages/data/src/components/use-select/index.js

```diff
@@ -134,6 +135,24 @@ function Store( registry, suspense ) {
 				listeningStores
 			);
 
+			if ( process.env.NODE_ENV === 'development' ) {
+				if ( ! didWarnUnstableReference ) {
+					const secondMapResult =
+						registry.__unstableMarkListeningStores(
+							() => mapSelect( select, registry ),
+							listeningStores
+						);
+
+					if ( ! isShallowEqual( mapResult, secondMapResult ) ) {
+						// eslint-disable-next-line no-console
+						console.warn(
+							`The values returned by the 'useSelect' hook aren't referentially stable. This can lead to unnecessary rerenders.`
```

Comment:
"Referentially stable" is an arcane term, who knows what it means 🙂 I think that more accessible version would be:
> The `useSelect` hook returns different values when called with the same state and parameters. This can lead to unnecessary rerenders.

Reply:
Sounds much better!